docs/decisions/0005-create-semantic-function.md

```diff
@@ -0,0 +1,141 @@
+1. Only supports prompt template config type of completion.
+    1. It’s not clear this restriction is meaningful or required. Developers can set this and still use a semantic function with an image generation model.
+1. The semantic function is created using the default template engine from the Kernel, it is added to the Kernels default skill collection and has the Kernels default text completion set on it. This links the semantic function with the kernel instance that was used to create it.
+    1. There is no option provided to have different life-cycles for semantic function versus kernel instances.
+    1. There is an assumption that all semantic functions will use the same template engine.
+1. The semantic function has a method called SetAIConfiguration. The parameter type taken by SetAIConfiguration is CompleteRequestSettings. When creating a semantic function. the CompleteRequestSettings are initialized from a CompletionConfig instance.
+    1. We have three terms i.e., AIConfiguration, CompleteRequestSettings and CompletionConfig referring to the same thing. We should standardize on one name.
+    1. The CompleteRequestSettings /CompletionConfig are very OpenAI specific s it's not possible to define the request settings used by other AI services.
+    1. A semantic function has only one CompleteRequestSettings /CompletionConfig instance which means it's not possible to associate model specific settings with a semantic function.
+1. Not clear why the PromptTemplateConfig does not contain the template, skill name and function name.
+1. The property DefaultServices is only referenced in the method PromptTemplateConfig.Compact(). This method also doesn’t seem to be referenced anywhere and its implementation is questionable because it has a side-effect of changing the instance in order to reduce JSON complexity.
+
+### Proposed Patterns
+
+#### 1. Simple case where semantic function lifecycle matches the semantic kernel and the default template engine is used.
+
+```csharp
+IKernel kernel = new KernelBuilder()
+    .WithAzureChatCompletionService(deploymentName: modelId, endpoint: endpoint, apiKey: apiKey)
+    .Build();
+
+var promptConfig = new PromptConfig()
+{
+    Template: "Say hello in German",
+    Description = "Say hello in German",
+    InputParameters = new(),
+    skillName: "GermanSkill",
+    functionName: "SayHello",
+    PromptRequestSettings = new List<PromptRequestSettings>() { … }
+};
+
+var function = kernel.CreateSemanticFunction(promptConfig);
+```
+
+
+#### Changes:
+
+The code looks similar to the current pattern but has the following key differences:
+
+1. Introduces the following new classes:
+    1. `PromptConfig` - Defines a prompt (aka semantic function) and the capabilities we want a prompt to support.
+    2. `PromptRequestSettings` - Defines the request settings used to execute the prompt with a specific model in an AI agnostic fashion.
+2. Mark `PromptTemplateConfig` and `SemanticFunctionConfig` as obsolete.
```

Comment:
Looks like in actual code these classes are not marked as obsolete, is there a plan to do it in follow-up PR?